src/crimson/os/seastore/btree/fixed_kv_btree.h

```diff
@@ -520,7 +520,9 @@ class FixedKVBtree {
             assert(prior.get_parent_node().get() == node.get());
             assert(node->children[i->get_offset()] == &prior);
           }
-        } else if (child_node->is_initial_pending()) {
+        } else if (child_node->is_initial_pending()
+                  || child_node->is_exist_clean()
+                  || child_node->is_exist_mutation_pending()) {
```

Comment:
should we include `is_exist_clean()` and `is_existing_mutation_pending()` in `is_initial_pending()`?

Reply:
Probably not as `is_initial_pending()`. But I don't see a good name to group them together.

Or use the following decision tree in the lba implementation, putting them in the else section:
* is_stable(): CLEAN_PENDING, CLEAN, DIRTY
* is_pending():
  - is_mutation_pending(): MUTATION_PENDING
  - else: INITIAL_WRITE_PENDING, EXIST_CLEAN, EXIST_MUTATION_PENDING
* else: !is_valid(): INVALID

What do you think?

Reply:
> Or use the following decision tree in the lba implementation, putting them in the else section:
> 
> * is_stable(): CLEAN_PENDING, CLEAN, DIRTY
> * is_pending():
>   
>   * is_mutation_pending(): MUTATION_PENDING
>   * else: INITIAL_WRITE_PENDING, EXIST_CLEAN, EXIST_MUTATION_PENDING
> * else: !is_valid(): INVALID

Sorry for the late reply, I think this is fine.

Reply:
@xxhdx1985126 @cyx1231st changed the decision path in new push ~